ducktape/cluster/remoteaccount.py

```diff
@@ -30,7 +61,99 @@ def __init__(self, hostname, user=None, ssh_args=None, ssh_hostname=None, extern
         self.ssh_args = ssh_args
         self.ssh_hostname = ssh_hostname
         self.externally_routable_ip = externally_routable_ip
-        self.logger = logger
+        self._logger = logger
+        self._ssh_config = None
+        self._ssh_client = None
+        self._sftp_client = None
+
+    @property
+    def logger(self):
+        if self._logger:
+            return self._logger
+        else:
+            return logging.getLogger(__name__)
+
+    @logger.setter
+    def logger(self, logger):
+        self._logger = logger
+
+    def _log(self, level, msg, *args, **kwargs):
+        msg = "%s: %s" % (str(self), msg)
+        self.logger.log(level, msg, *args, **kwargs)
+
+    @property
+    def ssh_config(self):
+        if not self._ssh_config:
+            self._ssh_config = self._parse_ssh_opts()
+        return self._ssh_config
+
+    @property
+    def ssh_client(self):
+        if not self._ssh_client:
+            o = self.ssh_config.lookup(self.hostname)
+
+            client = SSHClient()
+            client.set_missing_host_key_policy(AutoAddPolicy())
+
+            client.connect(
+                hostname=o.get('hostname', self.hostname),
+                port=int(o.get('port', 22)),
+                username=self.user,
+                password=None,
+                key_filename=o.get('identityfile'),
+                look_for_keys=False)
+            self._ssh_client = client
+
+        return self._ssh_client
+
+    @property
+    def sftp_client(self):
+        if not self._sftp_client:
+            self._sftp_client = self.ssh_client.open_sftp()
+
+        # TODO: can we check that it is still open before returning it? Not sure how timeouts would work for a long-lived session?
+        return self._sftp_client
+
+    def close(self):
+        """Close/release any outstanding network connections to remote account."""
+
+        if self._ssh_client:
+            self._ssh_client.close()
+            self._ssh_client = None
+        if self._sftp_client:
+            self._sftp_client.close()
+            self._sftp_client = None
+
+    def _parse_ssh_opts(self):
+        if self.ssh_args is None:
+            return SSHConfig()
+
+        args = self.ssh_args
```

Comment:
Doesn't necessarily have to happen here, but as long as we're moving away from the command line SSH, maybe we can make SSH args provided in a more flexible format from the Cluster implementations. In particular, I think we're actually undoing some of the construction of the ssh command that we do in VagrantCluster.